[개선] 엑셀값 검증

cabbage16 opened the pull request:
## 🎫 관련 이슈
[//]: # (다음 키워드를 사용하면 해당 PR을 머지할 때 자동으로 이슈를 닫을 수 있습니다.)
[//]: # (keyword: close|closes|closed|resolve|resolves|resolved|fix|fixes|fixed)
[//]: # (예시: close #1)

close #206

<br>

## 📄 개요
[//]: # (작업 내용을 간단히 요약해서 적습니다.)
[//]: # (예시: 유저 회원가입 기능을 만들었습니다.)

> 2차전형 점수 입력 시 오류 메시지를 더 상세화했습니다.

<br>

## 🔨 작업 내용
[//]: # (작업 내용을 자세하게 적습니다.)
[//]: # (붙임표 "-" 을 사용해서 목록을 만듭니다.)
[//]: # (예시: 유저 회원가입 API를 만들었습니다.)

- 엑셀 파일에 문제가 있는 경우, 어떤 셀에 어떤 값으로 인해 문제가 있는지 알려주도록 했습니다.
- InvalidFileException에서 어떤 이유로 인해 잘못된 파일인지 알려주도록 메시지를 수정했습니다.

<br>

## 🏁 확인 사항
[//]: # (PR을 보내기 전 다음 사항을 확인해주세요.)
[//]: # (해당 사항을 모두 이행해야 머지할 수 있습니다.)
[//]: # (- [x] 를 사용해서 완료로 표시할 수 있습니다.)

- [x] 테스트를 완료했나요?
- [x] API 문서를 작성했나요?
- [x] 코드 컨벤션을 준수했나요?
- [x] 불필요한 로그, 주석, import 등을 삭제했나요?

<br>

## 🙋🏻 덧붙일 말
[//]: # (다음 사항이 있다면 적어주세요.)
[//]: # (PR에 대한 추가 설명)
[//]: # (중점적으로 리뷰받고 싶은 부분)
[//]: # (기타 등등)

cabbage16 committed:
perf(#206): 2차전형 점수 입력 중 잘못된 셀을 에러에 표시
- 2차전형 점수 입력 엑셀에서 잘못된 값이 있을 경우, 몇행 몇열에 잘못된 값이 있는지 에러 메시지에서 알려주도록 수정했어요.

cabbage16 committed:
perf(#206): 잘못된 모든 셀을 한번에 에러메시지로 반환하도록 개선
- 행단위로 검사를 하면서 해당 행에 문제가 있으면 오류를 반환했지만, 오류가 있을 경우 리스트에 저장한 뒤 모두 조회를 한 후 한번에 반환하도록 개선했어요.

cabbage16 committed:
fix(#206): 엑셀 파일 점수 칸에 null 값이 들어있을 경우 서버 오류 해결
- Cell 이 null인것 과 cellType이 BLANK인 것은 서로 다른 문제여서 생기는 오류였습니다.
- 심층면접, NCS 점수 칸이 null인 경우 InvalidFileException이 발생하도록 했습니다.
- 코딩테스트 점수는 null일 수 있으므로 오류를 발생시키지 않고, 점수값을 추출할 때 null 체크를 하는 방식으로 변경했습니다.

cabbage16 committed:
test(#206): Exception 생성자 변경에 따른 테스트 수정
- InvalidFileException, WrongScoreException의 생성자가 변경되어서 테스트를 수정했어요.

cabbage16 committed:
test(#206): 2차전형점수 엑셀 파일 수정
- 수험번호 부여 규칙이 바뀌면서 테스트용 엑셀 파일을 변경했어요.

cabbage16 committed:
refactor(#206): 에러 메시지 수정

gimhanul commented on src/main/java/com/bamdoliro/maru/application/form/UpdateSecondRoundScoreUseCase.java:
```diff
@@ -51,79 +50,142 @@ public void execute(MultipartFile xlsx) throws IOException {
+                case "마이스터인재전형" -> {
+                    if (depthInterviewScore < 0 || depthInterviewScore > 120)
+                        wrongScoreCellList.add(depthInterviewScoreCell);
+                    if (ncsScore < 0 || ncsScore > 40)
```
소소한 팁인데
이런식으로 쓰면 가독성이 더 좋더라구욧

```suggestion
                    if (ncsScore < 0 || 40 < ncsScore)
```

```suggestion
                    if (!(0 <= ncsScore && ncsScore <= 40))
```

cabbage16 replied:
> 소소한 팁인데 이런식으로 쓰면 가독성이 더 좋더라구욧

저는 if 문 안에 되도록이면 긍정문을 넣고, 중첩 괄호를 지양하려고 했었는데
두번째처럼 하는 것은 실제 수학식 `0 <= ncsScore <= 40` 처럼 보이게 하기 위해서인가요?

gimhanul replied:
네 두번째는 그 의도입니당

gimhanul reviewed (approved):
그냥 기획적인 이야기인데,
이렇게 메시지 return하는 것보다 엑셀에 잘못된 셀 표시 + 메모로 뭐가 잘못되었는지 그려서 return 해주면 사용자가 더 사용하기 편할 거 같아요 ^_^

cabbage16 committed:
refactor(#206): 점수 검증식 변경
- 수학식처럼 보여서 가독성을 높이기 위해서 점수 범위 검증식을 변경했어요.

cabbage16 committed:
Merge branch 'develop' into perf/#206

cabbage16 committed:
Merge branch 'develop' into perf/#206

cabbage16 committed:
feat(#206): 2차 전형 점수 입력시 오류 표시 개선

- 잘못된 셀이 있으면 엑셀 파일 내부에 잘못된 셀 표시 후 어디가 잘못되었는지 메모하는 형식으로 오류를 출력합니다. (타입 불일치, 범위 초과)

cabbage16 committed:
test(#206): 2차 전형 점수 입력 테스트 수정

- 점수 입력 테스트 시 잘못된 타입과 범위 초과 케이스를 수정했습니다.

cabbage16 committed:
docs(#206): 2차 전형 점수 입력 테스트 수정

- 2차전형 점수 입력시 타입이 잘못되었거나, 점수 범위가 초과한 경우에 대한 응답 문서를 수정했습니다.

cabbage16 committed:
refactor(#206): WrongScoreException 삭제

- 점수 범위를 초과한 경우, WrongScoreException 대신 액셀 파일을 반환하므로 사용되지 않는 예외를 삭제했어요.
